controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php

```diff
@@ -416,47 +417,33 @@ function setPrimaryLocale($args, $request) {
 	//
 	// Helper methods.
 	//
-	/**
-	 * Update the locale support state (enabled or disabled).
-	 * @param $request Request
-	 * @param $rowId string The locale row id.
-	 * @param $enable boolean Enable locale flag.
-	 */
-	function _updateLocaleSupportState($request, $rowId, $enable) {
-		$newSupportedLocales = array();
-		$gridData = $this->getGridDataElements($request);
-
-		foreach ($gridData as $locale => $data) {
-			if ($data['supported']) {
-				array_push($newSupportedLocales, $locale);
-			}
-		}
-
-		if (AppLocale::isLocaleValid($rowId)) {
-			if ($enable) {
-				array_push($newSupportedLocales, $rowId);
-			} else {
-				$key = array_search($rowId, $newSupportedLocales);
-				if ($key !== false) unset($newSupportedLocales[$key]);
-			}
-		}
-
-		$site = $request->getSite();
-		$site->setSupportedLocales($newSupportedLocales);
-
-		$siteDao = DAORegistry::getDAO('SiteDAO');
-		$siteDao->updateObject($site);
-
-		$this->_updateContextLocaleSettings($request);
-	}
-
 	/**
 	 * Helper function to update locale settings in all
-	 * installed contexts, based on site locale settings.
+	 * installed journals, based on site locale settings.
 	 * @param $request object
 	 */
-	function _updateContextLocaleSettings($request) {
-		assert(false); // Must be implemented by subclasses
+	protected function _updateContextLocaleSettings($request) {
+		$site = $request->getSite();
+		$siteSupportedLocales = $site->getSupportedLocales();
+
+		$contextDao = Application::getContextDAO();
+		$contexts = $contextDao->getAll();
+		while ($context = $contexts->next()) {
+			$primaryLocale = $context->getPrimaryLocale();
+			foreach (array('supportedLocales', 'supportedFormLocales', 'supportedSubmissionLocales') as $settingName) {
+				$localeList = $context->getSetting($settingName);
+
+				if (isset($primaryLocale) && !in_array($primaryLocale, $siteSupportedLocales)) {
+					$context->setPrimaryLocale($site->getPrimaryLocale());
+					$this->updateContext($context);
+				}
```

Comment:
This if statement does not have to be in the foreach loop.
Also $this->updateContext does not exist, I think it should be $contextDao?

Reply:
I think that was existing code moved from elsewhere, but yes, I'll fix it.